### PR TITLE
Adds error handling for `mssql_idf` when module returns no matching results

### LINKED ISSUE #15059: Auxiliary failed: NoMethodError undefined method `each' for true:TrueClass
## Steps to reproduce

How'd you do it?

1. `use admin/mssql/mssql_idf`
2. `set RHOSTS <Target IP>`
3. `run`

## Expected behavior

Run auxiliary properly?

## Current behavior
```
msf6 auxiliary(admin/mssql/mssql_idf) > show options

Module options (auxiliary/admin/mssql/mssql_idf):

   Name                 Current Setting         Required  Description
   ----                 ---------------         --------  -----------
   NAMES                passw|bank|credit|card  yes       Pipe separated list of column names
   PASSWORD                                     no        The password for the specified username
   RHOSTS               172.26.0.24             yes       The target host(s), range CIDR identifier, or hosts file with syntax 'file:<path>'
   RPORT                1433                    yes       The target port (TCP)
   TDSENCRYPTION        false                   yes       Use TLS/SSL for TDS data "Force Encryption"
   USERNAME             sa                      no        The username to authenticate as
   USE_WINDOWS_AUTHENT  false                   yes       Use windows authentification (requires DOMAIN option set)

msf6 auxiliary(admin/mssql/mssql_idf) > run
[*] Running module against 172.26.0.24

[-] 172.26.0.24:1433 - Auxiliary failed: NoMethodError undefined method `each' for true:TrueClass
[-] 172.26.0.24:1433 - Call stack:
[-] 172.26.0.24:1433 -   /opt/metasploit-framework/embedded/framework/modules/auxiliary/admin/mssql/mssql_idf.rb:107:in `run'
[*] Auxiliary module execution completed
msf6 auxiliary(admin/mssql/mssql_idf) > 
```

### Metasploit version

6.0.41-dev-6ecaeaf7d7cdf410517824f757bb531e0735d782

### PR DESCRIPTION
**Pull request by cgranleese-r7**
Resolves #15059

This PR adds error handling for the scenario when a user attempts to run the `mssql_idf` module against a database that has no columns matching the pattern set within the `NAME` options within module options. 

Module options example:
 
image

This would cause the following error:
```
Auxiliary failed: NoMethodError undefined method `each' for true:TrueClass
```

This was due to `column_data` being equal to `nil`, as `result[:rows]` would only be generated if the module found results matching the `NAME` options search pattern. 

image

## Before 
image

## After 
image

## After with results being found
image

## Verification

- [ ] Start `msfconsole`
- [ ] `use admin/mssql/mssql_idf`
- [ ] `set` module options, my module options can be found above
- [ ] **Verify** the error is now caught and handled gracefully
- [ ] **Verify** the module no longer returns `Auxiliary failed: NoMethodError undefined method `each' for true:TrueClass`

## Database 
If wanting to set up the database to test that is all working as expected after the changes, I followed this [guide](https://docs.microsoft.com/en-us/sql/linux/quickstart-install-connect-docker?view=sql-server-ver15&pivots=cs1-bash).

Commands used to create the table: 
```SQL
CREATE TABLE CreditCard (id INT, CardNumber NVARCHAR(50))
INSERT INTO CreditCard VALUES (1, 1234567891011121);
INSERT INTO CreditCard VALUES (2, 1234567891011122);
INSERT INTO CreditCard VALUES (3, 1234567891011123);

# To verify all data was added use 
SELECT * FROM CreditCard
```

### CODE REVIEW

File: modules/auxiliary/admin/mssql/mssql_idf.rb [hunk 1]
```diff
@@ -104,11 +104,16 @@ def run
...
-        widths[col] = row[col].length if row[col].length > widths[col]
+    begin
+      (column_data|headings).each { |row|
+        0.upto(4) { |col|
+          widths[col] = row[col].length if row[col].length > widths[col]
+        }
       }
-    }
+    rescue NoMethodError
+      print_error("No columns matched the pattern #{datastore['NAMES'].inspect}. Update the NAMES option to change this search pattern.")
```
Comment (adfoster-r7) [inline modules/auxiliary/admin/mssql/mssql_idf.rb:114]: Would it be cleaner to have an explicit nil check for `column_data` earlier in the process, rather than catching an error after the fact? 👀

**Comment by adfoster-r7** [comment]
Since we're here it might be good to update the stale module description 😄 

>        The module has been tested against SQL Server 2005 but it should also work on
>        SQL Server 2008. The module will not work against SQL Server 2000 at this time,
>        if you are interested in supporting this platform, please contact the author.

File: modules/auxiliary/admin/mssql/mssql_idf.rb [hunk 2]
```diff
@@ -104,11 +104,16 @@ def run
...
+      return
+    end
 
     widths.each { |a|
```
Comment (adfoster-r7) [inline modules/auxiliary/admin/mssql/mssql_idf.rb:118]: This is scope creep, but a separate PR to update this to use rex-table might be nice

File: modules/auxiliary/admin/mssql/mssql_idf.rb (hunk 1 above)
Comment (cgranleese-r7) [reply modules/auxiliary/admin/mssql/mssql_idf.rb:114]: Yea, I can get that changed to be checked before it reaches that block that throws the error 👍

File: modules/auxiliary/admin/mssql/mssql_idf.rb (hunk 2 above)
Comment (cgranleese-r7) [reply modules/auxiliary/admin/mssql/mssql_idf.rb:118]: Sure, I'll look into that.

File: modules/auxiliary/admin/mssql/mssql_idf.rb [hunk 3]
```diff
@@ -22,9 +22,8 @@ def initialize(info = {})
         This module will search the specified MSSQL server for
         'interesting' columns and data.
 
-        The module has been tested against SQL Server 2005 but it should also work on
-        SQL Server 2008. The module will not work against SQL Server 2000 at this time,
-        if you are interested in supporting this platform, please contact the author.
+        This module has been tested against the latest SQL Server 2019 docker container image (22/04/2021).
+        If you are interested in supporting this platform, please contact the author.
```
Comment (adfoster-r7) [inline modules/auxiliary/admin/mssql/mssql_idf.rb:26]: This line doesn't quite make sense now, I believe they were suggesting to contact the author if you wanting to work on adding support for SQL Server 2000

Comment (cgranleese-r7) [reply modules/auxiliary/admin/mssql/mssql_idf.rb:26]: Oh ok, my bad, I see that now. 
Reading is hard :upside_down_face:

**Comment by adfoster-r7** [comment]
## Release Notes

Updates the Microsoft SQL Server interesting data finder module to correctly handle the scenario where no interesting data is found. Previously this would result in a module crash.
